x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/inference/results/AbstractDenseEmbeddingBitResults.java

```diff
@@ -0,0 +1,92 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ *
+ * this file was contributed to by a generative AI
+ */
+
+package org.elasticsearch.xpack.core.inference.results;
+
+import org.elasticsearch.common.io.stream.StreamInput;
+import org.elasticsearch.common.io.stream.StreamOutput;
+import org.elasticsearch.common.xcontent.ChunkedToXContentHelper;
+import org.elasticsearch.inference.InferenceResults;
+import org.elasticsearch.xcontent.ToXContent;
+import org.elasticsearch.xpack.core.ml.inference.results.MlDenseEmbeddingResults;
+
+import java.io.IOException;
+import java.util.Iterator;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+
+/**
+ * Abstract class from which other dense embedding bit result classes inherit their behaviour
+ */
+// Note: inheriting from AbstractDenseEmbeddingByteResults gives a bad implementation of the
+// Embedding.merge method for bits. TODO: implement a proper merge method
+public abstract class AbstractDenseEmbeddingBitResults implements DenseEmbeddingResults<AbstractDenseEmbeddingByteResults.Embedding> {
+    private final List<AbstractDenseEmbeddingByteResults.Embedding> embeddings;
+
+    public AbstractDenseEmbeddingBitResults(List<AbstractDenseEmbeddingByteResults.Embedding> embeddings) {
+        this.embeddings = embeddings;
+    }
+
+    public AbstractDenseEmbeddingBitResults(StreamInput in) throws IOException {
+        this(in.readCollectionAsList(AbstractDenseEmbeddingByteResults.Embedding::new));
+    }
+
+    public abstract String getArrayName();
```

Comment:
What do you think about storing the array field name as a member of the abstract class and the child classes can define it by passing a static string via a constructor (like `super(TEXT_EMBEDDING_BITS)`). Then we can make this method `getArrayName` a member of the abstract class (or we might not need it, I'm not sure if it's called externally).

Reply:
Yeah, I think this is better, since the array name doesn't need to be exposed outside of the abstract class.